source/rules.txt

```diff
@@ -0,0 +1,347 @@
+.. _mongodb-analyzer-rules:
+
+=========================
+Rules and Message Content
+=========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about the {+product+} rules and attached
+messages. A rule evaluates different parts of your code and returns a
+message if a specific condition is met.
```

Comment:
I was poking around the [mongo-csharp-analyzer](https://github.com/mongodb/mongo-csharp-analyzer/) repo to see if there was a good top level folder of test cases or source code that we could link to from here. It doesn't look like there is, or that at the very least, the user would need to match the rule name with the diagnostic test name, which is kind of messy.

So I couldn't find what I was looking for, but if you know something I don't, link it here. Otherwise, disregard this.

Reply:
Ya, i found that as well, but it mostly just tells you want to search for in the rest of the source code. The descriptors aren't super helpful (as far as I can tell) if you're trying to understand how the analyzer works.

I think the doc is find without it.